pandas/core/dtypes/missing.py

```diff
@@ -128,7 +131,9 @@ def _isna_ndarraylike(obj):
     values = getattr(obj, 'values', obj)
     dtype = values.dtype
 
-    if is_string_dtype(dtype):
+    if isinstance(values, ExtensionArray):
```

Comment:
Any reason not to duck-type this to check for objects that implement `isna` method?  I had a use case a while back where it would have been nice.

Reply:
I have a slight preference to using the interface here, instead of duck-typing, just since it's part of the interface (and I think `isna` is important enough to make it mandatory).

Virtual subclasses can register themselves if they want `pd.isna` to work, but don't want to inherit from `ExtensionArray`.